### PR TITLE
Add coverage for helper functions

### PR DESCRIPTION
**Pull request by cbowman0**
* Add input chekcs to safePow and safeStdDev
* Fix safeMap to work safely
* Add tests for safeSum()
* Add tests for safeDiff()
* Add tests for safeLen()
* Add tests for safeDiv()
* Add tests for safePow()
* Add tests for safeMul()
* Add tests for safeSubtract()
* Add tests for safeAvg()
* Add tests for safeStdDev()
* Add tests for safeLast()
* Add tests for safeMin()
* Add tests for safeMax()
* Add tests for safeAbs()
* Add tests for safeMap()
* Add tests for gcd()
* Add tests for lcm()
* Add/expand tests of normalize()
* Add tests for matchSeries()
* Add tests for formatPathExpressions()
* Add tests for sumSeries()
* Add tests for sumerSeriesWithWildcards()
* Add tests for averageSeriesWithWildcards()
* Add tests for diffSeries()
* Add tests for averageSeries()

### CODE REVIEW

File: webapp/graphite/render/functions.py [hunk 1]
```diff
@@ -110,7 +112,7 @@ def safeMax(values):
 def safeMap(function, values):
   safeValues = [v for v in values if v is not None]
   if safeValues:
-    return [function(x) for x in values]
+    return [function(x) for x in safeValues]
```
Comment (obfuscurity) [inline webapp/graphite/render/functions.py]: Nice catch here.

File: webapp/tests/test_functions.py [hunk 2]
```diff
@@ -20,6 +20,439 @@ def return_less(series, value):
...
+        self.assertEqual(functions.normalize([seriesList]), (seriesList, 0, 5, 1))
+
+    def test_normalize_generate_series_list_input(self):
+        seriesList = self._generate_series_list()
+        self.assertEqual(functions.normalize([seriesList]), (seriesList, 0, 101, 1))
+
+    #
+    # Test matchSeries()
+    #
+#def matchSeries(seriesList1, seriesList2):
```
Comment (obfuscurity) [inline webapp/tests/test_functions.py]: Is this still a WIP?

Comment (cbowman0) [inline webapp/tests/test_functions.py]: It wasn't intended to be, but that's definitely not done.  I'll clean it up tomorrow during my train ride.  Sorry for the extra garbage.

Comment (obfuscurity) [inline webapp/tests/test_functions.py]: No need to apologize, your work here is greatly appreciated.

**Comment by obfuscurity** [comment]
LGTM otherwise. Would be nice to get another set of 👀 on it though.

**Comment by cbowman0** [comment]
OK.  I'm done with correcting those remaining items.  Can someone else look it over?

**Comment by bmhatfield** [comment]
I did not check that your assertions are correct, but I skimmed over the rest and it all looks great to me. In particular, as before, I love the bug fixes that are a result of you adding coverage to things. Really great work here.

**Comment by obfuscurity** [comment]
Yeah I think this is a safe merge. The other similar PR (for glyph.py) probably demands more scrutiny because... it's glyph.py. 😛

**Comment by obfuscurity** [comment]
https://github.com/graphite-project/graphite-web/pull/1548 was the one I was thinking of.

**Comment by cbowman0** [comment]
Yes.  Agreed on extra people checking out #1548 .  That's some code that others should understand.

**Pull request merged**